[Unhandled Exception]: System.ObjectDisposedException: This object is no longer valid because the MetadataLoadContext that created it has been disposed.

### Issue Description

[Log](https://helixr1107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-111152-merge-29f1d7d5f1b74e3cb8/Workloads-ST-Wasm.Build.Tests.PInvokeTableGeneratorTests/1/console.907215bb.log?helixlogtype=result)
```
[] MSBUILD : error MSB4166: UNHANDLED EXCEPTIONS FROM PROCESS 1842:
[] MSBUILD : error MSB4166: =====================
[] MSBUILD : error MSB4166: 1/7/2025 5:29:23 PM
[] MSBUILD : error MSB4166: System.ObjectDisposedException: This object is no longer valid because the MetadataLoadContext that created it has been disposed.
[] MSBUILD : error MSB4166:    at System.Reflection.MetadataLoadContext.DisposeCheck()
[] MSBUILD : error MSB4166:    at System.Reflection.TypeLoading.Ecma.EcmaDefinitionType.ComputeDeclaringType()
[] MSBUILD : error MSB4166:    at System.Reflection.TypeLoading.RoType.get_DeclaringType()
[] MSBUILD : error MSB4166:    at System.Reflection.TypeLoading.RoDefinitionType.ComputeFullName()
[] MSBUILD : error MSB4166:    at System.Reflection.TypeLoading.RoType.get_FullName()
[] MSBUILD : error MSB4166:    at Microsoft.Build.Framework.LazyFormattedBuildEventArgs.WriteToStream(BinaryWriter writer)
[] MSBUILD : error MSB4166:    at Microsoft.Build.Framework.BuildMessageEventArgs.WriteToStream(BinaryWriter writer)
[] MSBUILD : error MSB4166:    at Microsoft.Build.Shared.LogMessagePacketBase.WriteToStream(ITranslator translator)
[] MSBUILD : error MSB4166:    at Microsoft.Build.BackEnd.NodeEndpointOutOfProcBase.RunReadLoop(Stream localReadPipe, Stream localWritePipe, ConcurrentQueue`1 localPacketQueue, AutoResetEvent localPacketAvailable, AutoResetEvent localTerminatePacketPump)
[] MSBUILD : error MSB4166: ===================
```

### Steps to Reproduce

I don't know how to reproduce.
It happened only once AFAIK during `Wasm.Build.Tests.PInvokeTableGeneratorTests` of `dotnet/runtime` build. 

Since it happened on Helix machine I also don't have the detailed log.

```
mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-22.04-helix-webassembly on a002M0D

.NET SDK Version: 10.0.100-alpha.1.24622.2
```

### Actual Behavior

No exception

### Analysis

_No response_

### Versions & Configurations

_No response_

--- a/src/Framework/LazyFormattedBuildEventArgs.cs
+++ b/src/Framework/LazyFormattedBuildEventArgs.cs
@@ -117,7 +117,17 @@ namespace Microsoft.Build.Framework
                     // Arguments may be ints, etc, so explicitly convert
                     // Convert.ToString returns String.Empty when it cannot convert, rather than throwing
                     // It returns null if the input is null.
-                    writer.Write(Convert.ToString(argument, CultureInfo.CurrentCulture) ?? "");
+                    string argValue;
+                    try
+                    {
+                        argValue = Convert.ToString(argument, CultureInfo.CurrentCulture) ?? "";
+                    }
+                    // Let's grace handle case where custom ToString implementation (that Convert.ToString fallbacks to) throws.
+                    catch (Exception e)
+                    {
+                        argValue = $"Argument conversion to string failed{Environment.NewLine}{e}";
+                    }
+                    writer.Write(argValue);
                 }
             }
             else

